.github/workflows/validation.yml

```diff
@@ -0,0 +1,54 @@
+on:
+  pull_request:
+    branches:
+      - '**'
+  push:
+    branches:
+      - master
+    tags:
+      - 'v**'
+
+name: validation
+jobs:
+  validate:
+    name: Linting
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout the repository
+        uses: actions/checkout@master
+      - name: Run validation
+        run: make validate
+  unit-testing:
+    name: Unit Testing
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout the repository
+        uses: actions/checkout@master
+
+      - name: Create artifacts directory
+        run: mkdir artifacts
+
+      - name: Run validation
+        run: make test 2>&1 | tee artifacts/unittest.log
```

Comment:
Should you `set -o pipefail`?
Usually it's off in which case the command won't necessarily fail in case `make test` fails, since the `tee` is the last thing to return so it's return code will be considered (unless you've set pipefail beforehand)

Reply:
oh true!